tests/cluster/tests/22-module-block.tcl

```diff
@@ -0,0 +1,110 @@
+#Check no permanent blocking when a client is blocked by module commands and a resharding happens at the same time.
+
+source "../tests/includes/init-tests.tcl"
+source "../../../tests/support/cli.tcl"
+
+set testmodule [file normalize ../../../tests/modules/blockonkeys.so]
+
+test "Create a 5 nodes cluster" {
+    create_cluster 5 5 
+}
+
+test "Cluster is up" {
+    assert_cluster_state ok
+}
+
+test "Cluster is writable" {
+    cluster_write_test 0
+}
+
+test "Load module in all the instances" {
+
+    foreach_redis_id id {
+        R $id module load $testmodule
+    }
+
+}
+
+set srcid -1
+set blockpid -1
+test "Exec module block cmd" {
+
+    exec \
+        ../../../src/redis-cli -c -h\
+        127.0.0.1 -p [get_instance_attrib redis 0 port] \
+	del k1 >>/dev/null
+
+    # Exec module block command
+    # k1 is mapped to slot 12706
+    set blockpid [exec \
+        ../../../src/redis-cli -c -h\
+        127.0.0.1 -p [get_instance_attrib redis 0 port] \
+	fsl.bpop k1 0 >>/dev/null & ]
```

Comment:
We have deferring clients for that, am I missing anything?

Reply:
AFAIK, the deferring client seem can't handle command redirection. Am i right?